lua/telescope/builtin/files.lua

```diff
@@ -289,134 +289,120 @@ local function prepare_match(entry, kind)
+    results_title = opts.files and Path:new(opts.cwd):make_relative(cwd) .. os_sep or "Results",
+    finder = fb_finder(opts),
     previewer = conf.file_previewer(opts),
     sorter = conf.file_sorter(opts),
     attach_mappings = function(prompt_bufnr, map)
       action_set.select:replace_if(function()
-        return is_dir(action_state.get_selected_entry().path)
+        -- test whether selected entry is directory
+        return action_state.get_selected_entry().path:sub(-1, -1) == os_sep
       end, function()
-        local new_cwd = vim.fn.expand(action_state.get_selected_entry().path:sub(1, -2))
+        local path = vim.loop.fs_realpath(action_state.get_selected_entry().path)
         local current_picker = action_state.get_current_picker(prompt_bufnr)
-        current_picker.cwd = new_cwd
-        current_picker:refresh(opts.new_finder(new_cwd), { reset_prompt = true })
+        current_picker.results_border:change_title(Path:new(path):make_relative(cwd) .. os_sep)
+        local finder = current_picker.finder
+        finder.files = true
+        finder.path = path
+        current_picker:refresh(finder, { reset_prompt = true, multi = current_picker._multi })
       end)
-
-      local create_new_file = function()
-        local current_picker = action_state.get_current_picker(prompt_bufnr)
-        local file = action_state.get_current_line()
-        if file == "" then
-          print(
-            "To create a new file or directory(add "
-              .. os_sep
-              .. " at the end of file) "
-              .. "write the desired new into the prompt and press <C-e>. "
-              .. "It works for not existing nested input as well."
-              .. "Example: this"
-              .. os_sep
-              .. "is"
-              .. os_sep
-              .. "a"
-              .. os_sep
-              .. "new_file.lua"
-          )
-          return
-        end
-
-        local fpath = current_picker.cwd .. os_sep .. file
-        if not is_dir(fpath) then
-          actions.close(prompt_bufnr)
-          Path:new(fpath):touch { parents = true }
-          vim.cmd(string.format(":e %s", fpath))
-        else
-          Path:new(fpath:sub(1, -2)):mkdir { parents = true }
-          local new_cwd = vim.fn.expand(fpath)
-          current_picker.cwd = new_cwd
-          current_picker:refresh(opts.new_finder(new_cwd), { reset_prompt = true })
-        end
-      end
-
-      map("i", "<C-e>", create_new_file)
-      map("n", "<C-e>", create_new_file)
+      map("i", "<C-e>", actions.create_file)
+      map("n", "<C-e>", actions.create_file)
+      map("i", "<C-r>", actions.rename_file)
+      map("n", "<C-r>", actions.rename_file)
+      map("i", "<C-y>", actions.copy_file)
+      map("n", "<y>", actions.copy_file)
+      map("i", "<C-h>", actions.toggle_hidden)
+      map("n", "<h>", actions.toggle_hidden)
+      map("i", "<C-g>", actions.goto_prev_dir)
+      map("n", "g", actions.goto_prev_dir)
```

Comment:
this is a bit slow when the user has g.. mappings in normal mode, it would be cool if somehow `<nowait>` is given (not sure that would fix it though)